### PR TITLE
Added app insights parameters to ARM templates

### LINKED ISSUE #81: Make App Insights work for the main application
An App Insights resource is created in the Azure Resource Group, but it seems nothing is hooked up to it yet. Make it work for at least the main application and preferably also for everything else of course 🙂 

Supporting docs: https://github.com/Microsoft/ApplicationInsights-aspnetcore/wiki/Getting-Started-with-Application-Insights-for-ASP.NET-Core

### PR DESCRIPTION
**Pull request by jfversluis**
Mostly have fixed values, just the instrumentation key is a parameter to be filled in at the build. Also, new settings are (partly) a slot setting because of reasons. Else they won't work.

Fixes #81

### CODE REVIEW

File: Deployment/complete-solution.json [hunk 1]
```diff
@@ -206,7 +206,21 @@
...
+            "TwitterOAuthTokenSecret": "[parameters('TwitterOAuthTokenSecret')]",
+            "APPINSIGHTS_PORTALINFO": "ASP.NETCORE"
+          }
+        },
+        {
+          "name": "appsettings",
+          "type": "config",
+          "apiVersion": "2015-08-01",
+          "dependsOn": [
+            "[resourceId('Microsoft.Web/Sites/', parameters('webSiteName'), 'production')]"
```
Comment (Jandev) [inline Deployment/complete-solution.json]: Are you sure these only get set in the Production slot?
I thought the notation was a bit different for slot settings, but don't have any examples available at this time.

**Comment by Jandev** [comment]
Check my comment for the slot setting. Otherwise looks pretty good.

File: Deployment/complete-solution.json (hunk 1 above)
Comment (jfversluis) [reply Deployment/complete-solution.json]: That is why I wanted you to review :P

For some weird reason App Insights will only work if it is a slot setting (ask me how I know). I Googled a bit and found this, for instance here: https://stackoverflow.com/questions/40332945/deployment-slot-specific-appsettin-in-arm-template

What do you think?

[the remark this answers is not in the document]
Comment (jfversluis) [reply Deployment/complete-solution.json]: Sure! Take your time, no rush. It works for now :)

[the remark this answers is not in the document]
Comment (Jandev) [reply Deployment/complete-solution.json]: I've searched through one of my local sideprojects and found some JSON which might help.
```
[
// Lots of other resources    
{
    "apiVersion": "2015-08-01",
    "name": "slotconfignames",
    "type": "config",
    "dependsOn": [
        "[resourceId('Microsoft.Web/Sites', parameters('myWebAppName'))]"
    ],
    "properties": {
        "appSettingNames": ["WEBJOBS_STOPPED"]
    }
},
{
    "apiVersion": "2016-08-01",
    "name": "staging",
    "type": "slots",
    "location": "westeurope",
    "dependsOn": [
        "[resourceId('Microsoft.Web/sites', parameters('myWebAppName'))]"
    ],
    "properties": {
        "enabled": true,
        "serverFarmId": "[resourceId('Microsoft.Web/serverfarms', parameters('myServiceplanName'))]",
        "siteConfig": {
            "netFrameworkVersion": "v4.7",
            "phpVersion": "",
            "pythonVersion": "",
            "nodeVersion": "",
            "linuxFxVersion": "",
            "requestTracingEnabled": false,
            "remoteDebuggingEnabled": false,
            "httpLoggingEnabled": false,
            "use32BitWorkerProcess": false,
            "webSocketsEnabled": false,
            "alwaysOn": true,
            "managedPipelineMode": "Integrated",
            "http20Enabled": true,
            "minTlsVersion": "1.2",
        },
        "clientAffinityEnabled": false,
        "clientCertEnabled": true,
        "httpsOnly": true
    },
    "resources": [{
        "name": "appsettings",
        "type": "config",
        "apiVersion": "2015-08-01",
        "dependsOn": [
            "[concat(resourceId('Microsoft.Web/sites', parameters('myWebAppName')), '/slots/staging')]"
        ],
        "tags": {
            "displayName": "AppServiceSettings"
        },
        "properties": {
            // All of your settings you need in the staging slot.
        }
    }]
}]
```

What you see over here is me adding a `slotconfignames` which is necessary to specify which appsettings are slot settings, like for example `WEBJOBS_STOPPED`.
Further on you'll see the type `slots`, which looks a lot like a normal App Service, just a different type and with some different dependencies.

Hope it helps!

**Comment by Jandev** [comment]
I guess this one is still in development because I'm still seeing #10 happen. Probably because the Image downloading isn't working somehow and we don't have any Insights yet?

**Comment by jfversluis** [comment]
I was just starting to look into this again. This stalled because I couldn't find the time to add the AppInsights slot setting the right way. Hopefully, I can look into that today and get the deploys up and running. Then I can start adding some bugfixes and features again :)

As for the App Insights and not working image downloading. It seems to still not work indeed, although AppInsights is working I don't see anything in particular there, unfortunately...

**Comment by jfversluis** [comment]
Hm yeah, test deployment fails because it now hard references to `https://raw.githubusercontent.com/jfversluis/CfpExchange/master/Deployment/` for the main template which doesn't include the new parameter. We might want to look into that at some point

**Comment by jfversluis** [comment]
Awesome, it works! 

screenshot 2019-01-18 10 33 32

Going for PRD deployment...

**Comment by Jandev** [comment]
#ShipIt

**Pull request merged**